docker/Dockerfile.bionic_debug

```diff
@@ -7,16 +7,16 @@ FROM ubuntu:bionic
 # sudo apt-get update && apt-get whichever
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
-      curl \
-      iptables \
-      iproute2 \
-      iputils-ping \
-      knot-dnsutils \
-      netcat \
-      tcpdump \
-      net-tools \
-      lsof \
-      sudo && \
+      curl=7.58.0-2ubuntu3.7 \
```

Comment:
Absolutely!! I will go ahead and put in some comments in here as to why we decided to pin versions. Thanks @howardjohn.

Reply:
@johnma14 I think this is key.   Is there a way we can enforce these as minimum versions but continue to fetch later versions as they are released?

Reply:
@duderino Using `apt-get install`, there isn't a way to do that as far as I know. 

@fpesce I understand the concern. With respect to the Dockerfile.bionic_debug, apart from pinning the versions, the only other difference with the older version is the `apt-get upgrade` command. That cleared up the security vulnerabilities with that one. With respect to the Dockerfile.xenial_debug, it has the `apt-get update && apt-get upgrade` commands but it was still not upgrading the packages to the latest version. So I ran some tests in my local env to build images using my private docker hub repo and it was building fine upgrading the packages correctly. I had mentioned this earlier in one of my comments (got buried :)) that I believe that the istio components being built is using an older istionightly/base_debug
 image which is available in the local repository which is why even thought the istio component versions are new, it is still showing these vulnerabilities related older versions of base packages in the istionightly/base_debug. 

We can try doing a `docker build --pull` when building these base OS docker images instead of just calling `docker build` command. I made similar changes to the sample bookinfo images: https://github.com/istio/istio/pull/13716 which got merged but we haven't build any new images since then.